guide/self_hosted_registry.md

```diff
@@ -0,0 +1,159 @@
+# Using a self-hosted registry with OpenFaaS
+
+If you're using OpenFaaS on single host, then you don't need to push your images to a registry. They will just be used from the local Docker library.
+
+If you are using a remote server or a multi-node cluster then you can push your function's image to a registry or the Docker Hub.
+
+This describes how to use OpenFaaS in a swarm with your own local registry for hosting function images.
```

Comment:
Third option - private images/repos. #87 #83

Reply:
I think I cover this with the pointer to https://github.com/openfaas/faas/blob/master/docs/managing-images.md#deploy-your-own-private-docker-registry later in this document.